lib/client.js

```diff
@@ -0,0 +1,169 @@
+var WS = require('ws');
+var debug = require('debug')('docker-exec-websocket-server:lib:client');
+var url = require('url');
+var events = require('events');
+var _ = require('lodash');
+var assert = require('assert');
+var msgcode = require('../lib/messagecodes.js');
+var through = require('through');
+
+export default class DockerExecWebsocketClient extends events.EventEmitter {
+  constructor (options) {
+    super();
+    this.options = _.defaults({}, options, {
+      tty: true,
+      command: ['/bin/bash'],
+      protocol: 'ws',
+      slashes: true,
+    });
+  }
+
+  /* Makes a client program with unbroken stdin, stdout, stderr streams
+   * Is also an EventEmitter with 'exit' event
+   *
+   * Required options:
+   * url parts (hostname port pathname)
+   * or url
+   * tty: whether or not we expect VT100 style output
+   * command: array or string of command to be run in exec
+   * not supported yet: ssl cert
+   */
+  async execute () {
+    this.options.query = {
+      tty: encodeURIComponent(this.options.tty ? 'true' : 'false'),
+      command: this.options.command,
+    };
+    this.url = this.options.url || url.format(this.options);
+    assert(this.url, 'url required or malformed url input');
+
+    this.socket = new WS(this.url);
+    this.socket.binaryType = 'arraybuffer';
+    this.socket.onopen = () => {
+      console.log('socket opened');
+      this.emit('open');
+    };
+
+    //set state, state does nothing yet
+    this.state = msgcode.pause;
+    //stdin stream with pause buffering
+    this.stdin = through((data) => {
+      this.sendMessage(msgcode.stdin, data);
+    });
+
+    this.bufin = through(function (data) {
+      console.log(data);
```

Comment:
You can use `debug` in the browser too...
But maybe add a static property like this:
```js
static debug = require('debug').enable(pattern);
```

Then in browser we can do
```js
DockerExecWebsocketClient.debug.enable("*"); // to enable
DockerExecWebsocketClient.debug.disable(); // to disable
```

Debug remembers between reloads of the page see: https://github.com/visionmedia/debug#browser-support